MonoGame.Framework/Graphics/SpriteBatcher.cs

```diff
@@ -163,14 +163,14 @@ private void EnsureArrayCapacity(int numBatchItems)
         }
 
         /// <summary>
-        /// Reference comparison of the underlying Texture objects for each given SpriteBatchitem.
+        /// Comparison of the underlying Texture objects for each given SpriteBatchitem.
         /// </summary>
         /// <param name="a"></param>
         /// <param name="b"></param>
-        /// <returns>0 if they are not reference equal, and 1 if so.</returns>
+        /// <returns>0 if they are equal, -1 or 1 if not.</returns>
 	    static int CompareTexture ( SpriteBatchItem a, SpriteBatchItem b )
 		{
-            return ReferenceEquals( a.Texture, b.Texture ) ? 0 : 1;
+            return a.Texture.ComparisonCode.CompareTo(b.Texture.ComparisonCode);
```

Comment:
We should consider making `Texture` inherit from [IComparable<T>](https://msdn.microsoft.com/en-us/library/4d7sx9hd(v=vs.110).aspx).  This way we solve this problem for us in `SpriteBatch` and give end users a way to do the same type of sorting in their own code without having to solve this issue themselves.